packages/backend-app-api/src/http/applyInternalErrorFilter.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { LoggerService } from '@backstage/backend-plugin-api';
+import { assertError } from '@backstage/errors';
+import { randomBytes } from 'crypto';
+
+function handleBadError(error: Error, logger: LoggerService) {
+  const logId = randomBytes(10).toString('hex');
+  logger.error(
```

Comment:
would make sense to have the log id as meta info so it ends up in json formatted logs as a dedicated field right

Reply:
Yeah would be nice, but we can't have both that and the error fields being expanded. Might not be what we want though tbh, as in errors being expanded. I'll have a look at what a middle ground could be

Reply:
Alright got it in there with a child logger